android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java

```diff
@@ -220,6 +220,14 @@ public int getENSResolveMethod() {
         return BravePrefServiceBridgeJni.get().getENSResolveMethod();
     }
 
+    public void setSnsResolveMethod(int method) {
+        BravePrefServiceBridgeJni.get().setSnsResolveMethod(method);
+    }
+
+    public int getSnsResolveMethod() {
+        return BravePrefServiceBridgeJni.get().getSnsResolveMethod();
+    }
```

Comment:
Please avoid using `BravePrefServiceBridge` at all. We are going to retire it soon. We should use upstream's `UserPrefs` instead. Check that as an example https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/java/src/org/chromium/chrome/browser/sync/settings/AccountManagementFragment.java;l=289?q=AccountManagementFragment.java&ss=chromium%2Fchromium%2Fsrc.
That is a place on where you add you cpp header enum https://github.com/brave/brave-core/blob/master/build/android/config.gni#L56 and that's where you populate your preference https://github.com/brave/brave-core/blob/master/browser/android/preferences/BUILD.gn#L35. After all of that you can access your pref from java similar to this `UserPrefs.get(Profile.getLastUsedRegularProfile())                                        .getBoolean(BravePref.USE_REWARDS_STAGING_SERVER)`

Reply:
Well I just copypasted from similar methods above :)